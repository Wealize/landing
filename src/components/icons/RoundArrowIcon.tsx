import React from 'react'

const RoundArrowIcon = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="6em"
      height="6em"
      viewBox="0 0 90 90"
      {...props}
    >
      <path fill="#66f" d="M44.999 89.999a45 45 0 1145-45 45 45 0 01-45 45z" />
      <path
        fill="#fff"
        d="M32.513 36.173l11.2-9.05v38.525h2.119V27.162l11.158 9.012 1.336-1.654-12.493-10.09-.415-.335h-1.336l-.368.3-12.54 10.129z"
      />
    </svg>
  )
}

export default React.memo(RoundArrowIcon)
