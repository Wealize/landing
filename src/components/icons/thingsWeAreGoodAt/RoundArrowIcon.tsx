import React from 'react'

const RoundArrowIcon = (props: React.SVGProps<SVGSVGElement>) => {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="100%"
      height="100%"
      viewBox="0 0 40 40"
      {...props}
    >
      <g fill="none">
        <path d="M20 40a20 20 0 1120-20 20 20 0 01-20 20z" />
        <path
          fill="#f06078"
          d="M20 38.5c4.942 0 9.587-1.924 13.081-5.419A18.375 18.375 0 0038.5 20c0-4.942-1.924-9.587-5.419-13.081A18.375 18.375 0 0020 1.5 18.374 18.374 0 006.919 6.919 18.375 18.375 0 001.5 20c0 4.942 1.924 9.587 5.419 13.081A18.375 18.375 0 0020 38.5m0 1.5C8.954 40 0 31.046 0 20S8.954 0 20 0s20 8.954 20 20-8.954 20-20 20z"
        />
      </g>
      <path
        fill="#f06078"
        d="M13.79 15.465l5.4-4.139v19.673h1.5V11.325l5.4 4.139.912-1.19-6.311-4.839-.045-.035h-1.41l-.045.034-6.312 4.839z"
      />
    </svg>
  )
}

export default React.memo(RoundArrowIcon)
