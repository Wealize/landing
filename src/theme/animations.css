/* appear - on page load */
.fade-appear {
  opacity: 0;
  z-index: 1;
}
.fade-appear.fade-appear-active {
  opacity: 1;
  transition: opacity 1000ms linear;
}

/* enter */
.fade-enter {
  opacity: 0;
  z-index: 1;
}
.fade-enter.fade-enter-active {
  opacity: 1;
  transition: opacity 5000ms linear 5000ms;
}

/* exit */
.fade-exit {
  opacity: 1;
}
.fade-exit.fade-exit-active {
  opacity: 0;
  transition: opacity 5000ms linear;
}
.fade-exit-done {
  opacity: 0;
}

/* slide enter */
.slide-enter {
  opacity: 0;
  transform: scale(0.97) translateY(5px);
  z-index: 1;
}
.slide-enter.slide-enter-active {
  opacity: 1;
  transform: scale(1) translateY(0);
  transition: opacity 3000ms linear 1000ms, transform 3000ms ease-in-out 1000ms;
}

/* slide exit */
.slide-exit {
  opacity: 1;
  transform: scale(1) translateY(0);
}
.slide-exit.slide-exit-active {
  opacity: 0;
  transform: scale(0.97) translateY(5px);
  transition: opacity 1500ms linear, transform 1500ms ease-out;
}
.slide-exit-done {
  opacity: 0;
}
