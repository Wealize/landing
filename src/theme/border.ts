export const DEFAULT_RADIUS = 28
