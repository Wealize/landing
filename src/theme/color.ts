export const PRIMARY = '#f06078'
export const ACCENT = '#6666ff'
export const DARK_GRAY = '#3c3c3b'
export const LIGHT_GRAY = '#f5f5f5'

export const WHITE = '#ffffff'
export const BLACK = '#000000'
