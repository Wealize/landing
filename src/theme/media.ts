import { generateMedia } from 'styled-media-query'

export const BREAKPOINT_XS = '480px'
export const BREAKPOINT_SM = '576px'
export const BREAKPOINT_MD = '768px'
export const BREAKPOINT_LG = '992px'
export const BREAKPOINT_XL = '1366px'
export const BREAKPOINT_XXL = '1600px'
export const BREAKPOINT_XXXL = '1800px'

export const media = generateMedia({
  xs: BREAKPOINT_XS,
  sm: BREAKPOINT_SM,
  md: BREAKPOINT_MD,
  lg: BREAKPOINT_LG,
  xl: BREAKPOINT_XL,
  xxl: BREAKPOINT_XXL,
  xxxl: BREAKPOINT_XXXL
})
