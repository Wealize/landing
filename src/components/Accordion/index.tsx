import React, { useState } from 'react'
import useTranslation from 'next-translate/useTranslation'

import { AccordionProps, AccordionSectionData } from '../../interfaces/Accordion'
import ButtonRounded from '../Buttons/Rounded'
import { SMALL } from '../Buttons/Rounded/sizes'
import RoundArrowIcon from '../icons/thingsWeAreGoodAt/RoundArrowIcon'

import {
  SectionRowElementContainer,
  SectionElementContainer,
  Container,
  Separator,
  TitleContainer,
  Title,
  SectionContainer,
  SectionHeader,
  SectionTitle,
  SectionToggle,
  SectionContentContainer,
  SectionElement
} from './styles'
const Accordion = (props: AccordionProps) => {
  const {
    backgroundColor,
    textColor,
    title,
    sectionsData,
    ToggleButtonBackgroundColor,
    ToggleButtonTextColor,
    ToggleButtonAnimatedBackgroundColor,
    ToggleButtonAnimatedTextColor
  } = props

  const { t } = useTranslation('profile')

  return (
    <Container backgroundColor={backgroundColor} textColor={textColor}>
      <TitleContainer>
        <Title textColor={textColor}>{t('accordion.title', { text: title })}</Title>
        <Separator className="accordion-separator" textColor={textColor}/>
      </TitleContainer>

      {
        sectionsData.map((section: AccordionSectionData, index: number) => {
          const [isExpanded, setIsExpanded] = useState(index === 0)
          return (
          <SectionContainer
            className="accordion-section-container"
            key={index}
          >
            <SectionHeader isExpanded={isExpanded}>
              <SectionTitle>{t(`accordion.section-${index}.title`, { text: section.title })}</SectionTitle>
              <SectionToggle
                className="accordion-section-toggle-button"
                textColor={textColor}
                backgroundColor={backgroundColor}
                isExpanded={isExpanded}
                onClick={() => setIsExpanded(!isExpanded)}
                aria-label={`expand ${section.title} content button`}
              >
                <RoundArrowIcon />
              </SectionToggle>
              <ButtonRounded
                backgroundColor={ToggleButtonBackgroundColor}
                textColor={ToggleButtonTextColor}
                animatedBackgroundColor={ToggleButtonAnimatedBackgroundColor}
                animatedTextColor={ToggleButtonAnimatedTextColor}
                size={SMALL}
                handleClick={() => setIsExpanded(!isExpanded)}
              >
                <span>{isExpanded ? t('accordion.toogle-button-opened-text') : t('accordion.toogle-button-closed-text')}</span>
              </ButtonRounded>
            </SectionHeader>
            <SectionContentContainer isExpanded={isExpanded}>
              {section.data.map((element: string, index: number) => (
                <SectionElementContainer key={index}>
                  <SectionRowElementContainer>
                    <i> -&nbsp; </i>
                    <SectionElement>{element}</SectionElement>
                  </SectionRowElementContainer>
                </SectionElementContainer>

              ))}
            </SectionContentContainer>
            <Separator className="accordion-separator" textColor={textColor}/>
          </SectionContainer>
          )
        })
      }


    </Container>
  )
}

export default React.memo(Accordion)
