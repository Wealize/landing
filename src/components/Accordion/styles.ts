import styled from 'styled-components'

import { sixthPx, halfPx, twoThirdPx, singlePx, singleAndHalfPx, doublePx, doubleAndHalfPx, triplePx, quadruplePx, quadruple } from '../../theme/space'
import { media } from '../../theme/media'

export const Container = styled.div<{ backgroundColor: string, textColor: string}>`
  width: 100%;
  min-height: 20vh;
  padding: ${doubleAndHalfPx()} ${singlePx()};

  ${media.greaterThan('md')`
    padding: ${triplePx()} ${quadruplePx()};
  `};

  ${media.greaterThan('lg')`
    padding: ${quadruplePx()} ${quadruple() * 1.5}px;
  `};

  ${media.greaterThan('xl')`
    padding: ${quadruplePx()} ${quadruple() * 2.5}px;
  `};

  ${media.greaterThan('xxl')`
    padding: ${quadruplePx()} ${quadruple() * 3.5}px;
  `};

  color: ${({ textColor }) => textColor};
  background-color: ${({ backgroundColor }) => backgroundColor};
`

export const TitleContainer = styled.div`
  padding-bottom: 0;

  > .accordion-separator{
    margin-top: ${halfPx()};
  }
`

export const Title = styled.h5<{ textColor: string}>`
  font-size: 1rem;
  text-transform: uppercase;
  font-weight: bold;
  color: ${({ textColor }) => textColor};

  ${media.greaterThan('xl')`
    font-size: 1.1rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 1.2rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 1.2rem;
  `};
`

export const Separator = styled.hr<{ textColor: string}>`
  width: 100%;
  background-color: ${({ textColor }) => textColor};
  opacity: .6;
  border: none;
  height: 2px;
  margin: 0;
`

export const SectionContainer = styled.div`
  padding-top: ${singleAndHalfPx()};

  > .accordion-separator {
    margin-top: ${singleAndHalfPx()};
  }

  ${media.greaterThan('md')`
    padding-top: ${doublePx()};

    > .accordion-separator {
      margin-top: ${doublePx()};
    }
  `};
`

export const SectionHeader = styled.div<{isExpanded: boolean}>`
  display: flex;
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 0;

  > .button-rounded {
    display: none;

    & span {
      font-weight: lighter;
    }
  }


  ${media.greaterThan('lg')`
    > .accordion-section-toggle-button {
      display: none;
    }

    > .button-rounded {
      display: block;
      min-height: 75px;
    }
  `};

`

export const SectionTitle = styled.h6`
  font-size: 1.7rem;
  width: 75%;
  line-height: 1.52;

  ${media.greaterThan('md')`
    font-size: 1.8rem;
  `};

  ${media.greaterThan('lg')`
    font-size: 2.2em;
  `};

  ${media.greaterThan('xl')`
    font-size: 2.4rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 2.7rem;
  `};
`

export const SectionToggle = styled.button<{
  textColor: string
  backgroundColor: string
  isExpanded: boolean
}>`
  width: 40px;
  height: 40px;
  background-color: transparent;
  border: none;
  padding: 0;
  transition: all 0.5s ease-out;
  transform: ${({ isExpanded }) =>
    isExpanded ? 'rotate(0deg)' : 'rotate(180deg)'};

  & svg > g:first-child {
    fill: ${({ textColor, isExpanded }) => isExpanded ? textColor : 'none'};
  }

  & svg > g > path:last-child {
    fill: ${({ textColor }) => textColor};
  }

  & svg > path {
    fill: ${({ textColor, backgroundColor, isExpanded }) => isExpanded ? backgroundColor : textColor};
  }

  ${media.greaterThan('md')`
      width: 50px;
      height: 50px;
  `};

  ${media.greaterThan('lg')`
    display: none;
  `};
`

export const SectionContentContainer = styled.div<{isExpanded: boolean}>`
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-column-gap: ${singlePx()};
  transition: all 0.5s ease-out;
  overflow: hidden;
  height: auto;
  max-height: ${({ isExpanded }) => (isExpanded ? 1000 : 0)}px;
  padding-top: ${({ isExpanded }) => (isExpanded ? singleAndHalfPx() : 0)};

  ${media.greaterThan('lg')`
    grid-template-columns: repeat(3, 1fr);
  `};

  ${media.greaterThan('xl')`
    grid-template-columns: repeat(3, 1fr);
  `};

  ${media.greaterThan('xxl')`
    grid-template-columns: repeat(4, 1fr);
  `};

`

export const SectionElement = styled.span`
  padding: ${twoThirdPx()} 0 ${sixthPx()};
  line-height: 1.4;

  &:before {
    content: '- '
  }

  ${media.greaterThan('md')`
    font-size: 1.2rem;
    padding: ${twoThirdPx()};
  `};

  ${media.greaterThan('lg')`
    font-size: 1.3rem;
  `};

  ${media.greaterThan('xl')`
    font-size: 1.4rem;
  `};

  ${media.greaterThan('xxl')`
    font-size: 1.5rem;
  `};
`
