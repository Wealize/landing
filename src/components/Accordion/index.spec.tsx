import React from 'react'
import { shallow, mount } from 'enzyme'

import RoundArrowIcon from '../icons/thingsWeAreGoodAt/RoundArrowIcon'
import { WHITE_COLOR, ACCENT_COLOR } from '../../theme/color'

import Accordion from './index'
import {
  Separator,
  Title,
  SectionContainer,
  SectionTitle,
  SectionToggle,
  SectionElement
} from './styles'
describe('<Accordion />', () => {
  const sectionsData = [
    {
      title: 'Languages & frameworks',
      data: ['Cypress', 'Jest', 'Redux', 'Fabric', 'Hyperledger Besu', 'Quorum', 'web3']
    },
    {
      title: 'Devops, Architecture & Systems',
      data: ['Cypress', 'Jest', 'Redux', 'Fabric', 'Hyperledger Besu', 'Quorum', 'web3']
    }
  ]
  const accordionProps = {
    backgroundColor: ACCENT_COLOR,
    textColor: WHITE_COLOR,
    title: 'Tools & Tech Stack',
    sectionsData: sectionsData,
    ToggleButtonBackgroundColor: ACCENT_COLOR,
    ToggleButtonTextColor: WHITE_COLOR,
    ToggleButtonAnimatedBackgroundColor: WHITE_COLOR,
    ToggleButtonAnimatedTextColor: ACCENT_COLOR
  }
  const wrapper = shallow(<Accordion {...accordionProps} />)

  it('should render', () => {
    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render content correctly', () => {
    expect(wrapper.find(Separator).length).toEqual(3)
    expect(wrapper.find(SectionContainer).length).toEqual(2)
    expect(wrapper.find(Title).text()).toEqual(accordionProps.title)
    expect(wrapper.find(SectionTitle).length).toEqual(2)
    expect(wrapper.find(SectionTitle).at(0).text()).toEqual(sectionsData[0].title)
    expect(wrapper.find(SectionTitle).at(1).text()).toEqual(sectionsData[1].title)
    expect(wrapper.find(SectionToggle).length).toEqual(2)
    expect(wrapper.find(RoundArrowIcon).length).toEqual(2)
    expect(wrapper.find(SectionElement).length).toEqual(14)
  })

  it('should click on toggle button', () => {
    const setIsExpanded = jest.fn()
    const toggleComponent = mount(
      <SectionToggle
        onClick={setIsExpanded}
        textColor={WHITE_COLOR}
        backgroundColor={ACCENT_COLOR} isExpanded={false}
      />
    )
    const handleClick = jest.spyOn(React, 'useState')
    handleClick.mockImplementation(isExpanded => [isExpanded, setIsExpanded])
    toggleComponent.simulate('click')
    expect(setIsExpanded).toBeTruthy()
  })
})
