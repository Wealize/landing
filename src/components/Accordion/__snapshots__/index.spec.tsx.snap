// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion /> should render 1`] = `
<styles__Container
  backgroundColor="#6666ff"
  textColor="#ffffff"
>
  <styles__TitleContainer>
    <styles__Title
      textColor="#ffffff"
    >
      profile:accordion.title
    </styles__Title>
    <styles__Separator
      className="accordion-separator"
      textColor="#ffffff"
    />
  </styles__TitleContainer>
  <styles__SectionContainer
    className="accordion-section-container"
    key="0"
  >
    <styles__SectionHeader
      isExpanded={true}
    >
      <styles__SectionTitle>
        profile:accordion.section-0.title
      </styles__SectionTitle>
      <styles__SectionToggle
        aria-label="expand Languages & frameworks content button"
        backgroundColor="#6666ff"
        className="accordion-section-toggle-button"
        isExpanded={true}
        onClick={[Function]}
        textColor="#ffffff"
      >
        <Memo(RoundArrowIcon) />
      </styles__SectionToggle>
      <Memo(ButtonRounded)
        animatedBackgroundColor="#ffffff"
        animatedTextColor="#6666ff"
        backgroundColor="#6666ff"
        handleClick={[Function]}
        size={
          Object {
            "desktop": Array [
              170,
              50,
            ],
            "mobile": Array [
              90,
              50,
            ],
            "tablet": Array [
              150,
              50,
            ],
            "wide": Array [
              190,
              50,
            ],
          }
        }
        textColor="#ffffff"
      >
        profile:accordion.toogle-button-opened-text
      </Memo(ButtonRounded)>
    </styles__SectionHeader>
    <styles__SectionContentContainer
      isExpanded={true}
    >
      <styles__SectionElement
        key="0"
      >
        Cypress
      </styles__SectionElement>
      <styles__SectionElement
        key="1"
      >
        Jest
      </styles__SectionElement>
      <styles__SectionElement
        key="2"
      >
        Redux
      </styles__SectionElement>
      <styles__SectionElement
        key="3"
      >
        Fabric
      </styles__SectionElement>
      <styles__SectionElement
        key="4"
      >
        Hyperledger Besu
      </styles__SectionElement>
      <styles__SectionElement
        key="5"
      >
        Quorum
      </styles__SectionElement>
      <styles__SectionElement
        key="6"
      >
        web3
      </styles__SectionElement>
    </styles__SectionContentContainer>
    <styles__Separator
      className="accordion-separator"
      textColor="#ffffff"
    />
  </styles__SectionContainer>
  <styles__SectionContainer
    className="accordion-section-container"
    key="1"
  >
    <styles__SectionHeader
      isExpanded={false}
    >
      <styles__SectionTitle>
        profile:accordion.section-1.title
      </styles__SectionTitle>
      <styles__SectionToggle
        aria-label="expand Devops, Architecture & Systems content button"
        backgroundColor="#6666ff"
        className="accordion-section-toggle-button"
        isExpanded={false}
        onClick={[Function]}
        textColor="#ffffff"
      >
        <Memo(RoundArrowIcon) />
      </styles__SectionToggle>
      <Memo(ButtonRounded)
        animatedBackgroundColor="#ffffff"
        animatedTextColor="#6666ff"
        backgroundColor="#6666ff"
        handleClick={[Function]}
        size={
          Object {
            "desktop": Array [
              170,
              50,
            ],
            "mobile": Array [
              90,
              50,
            ],
            "tablet": Array [
              150,
              50,
            ],
            "wide": Array [
              190,
              50,
            ],
          }
        }
        textColor="#ffffff"
      >
        profile:accordion.toogle-button-closed-text
      </Memo(ButtonRounded)>
    </styles__SectionHeader>
    <styles__SectionContentContainer
      isExpanded={false}
    >
      <styles__SectionElement
        key="0"
      >
        Cypress
      </styles__SectionElement>
      <styles__SectionElement
        key="1"
      >
        Jest
      </styles__SectionElement>
      <styles__SectionElement
        key="2"
      >
        Redux
      </styles__SectionElement>
      <styles__SectionElement
        key="3"
      >
        Fabric
      </styles__SectionElement>
      <styles__SectionElement
        key="4"
      >
        Hyperledger Besu
      </styles__SectionElement>
      <styles__SectionElement
        key="5"
      >
        Quorum
      </styles__SectionElement>
      <styles__SectionElement
        key="6"
      >
        web3
      </styles__SectionElement>
    </styles__SectionContentContainer>
    <styles__Separator
      className="accordion-separator"
      textColor="#ffffff"
    />
  </styles__SectionContainer>
</styles__Container>
`;
